lib/pact_broker/client/colorize_notices.rb

```diff
@@ -11,16 +11,25 @@ def self.call(notices)
 
       def self.colorized_message(notice)
         color = color_for_type(notice.type)
-        if color
+        uri_string = ::URI.extract(notice.text, %w(http https))
+        if color && uri_string.count >= 1
+          color_for_url(::Term::ANSIColor.color(color, notice.text || ''), uri_string)
+        elsif color
           ::Term::ANSIColor.color(color, notice.text || '')
+        elsif uri_string.count >= 1
+          color_for_url(notice.text, uri_string)
         else
           notice.text
         end
       end
 
+      def self.color_for_url(text, uris)
+        text.gsub!(uris.first, ::Term::ANSIColor.magenta(uris.first))
```

Comment:
if `magenta` is still too dark, we can experiment here

Reply:
Also if we decide just not to colour the URL, that is okay :)

The changes, in `lib/pactflow/client/provider_contracts/publish.rb` should still be applicable without this, if we remove this method, we just remove the colouring of the url, and our nice informational messaging remains the same

Reply:
I have not tested this, but this will probably work. It will accumulate all the changes into one string.

```
uris.inject(text) do | new_text, uri  |
  new_text.gsub(uri, ::Term::ANSIColor.magenta(uris.first))
end
```

`gsub!` modifies in place, and good practice for a method that returns a value is to not modify the argument, so `gsub` is preferred.

Reply:
nice, just needed to change `:Term::ANSIColor.magenta(uri.first)` to `:Term::ANSIColor.magenta(uri)`

```
      def self.color_for_url(text, uris)
        uris.inject(text) do | new_text, uri  |
          new_text.gsub(uri, ::Term::ANSIColor.magenta(uri))
        end
      end
```

![Screenshot 2023-05-05 at 12 30 19](https://user-images.githubusercontent.com/19932401/236446598-89d9e984-15c2-444e-8507-8cbbfe651f83.png)